### PR TITLE
Beer favourite

### PR DESCRIPTION
**Pull request by matt-harris**
Added button to beer component to toggle is favourite.
Added button to App to filter your favourites.

Demo
punkapi.surge.sh

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -33,13 +33,16 @@
               <option value="strong">Strong (7.5% +)</option>
             </select>
           </BeerFilter>
+
+          <button v-show="!showFavourites" class="btn" @click="toggleFavourites">Show favourites</button>
```
Comment (atrunelle) [inline src/App.vue]: This is repetitive. If the only difference is the text, you can use a computed property to display the text you want depending on `showFavourites` value

File: src/App.vue [hunk 2]
```diff
@@ -66,9 +69,15 @@ export default {
...
       errors: [],
       beerSearch: '',
-      beerStrength: 'all'
+      beerStrength: 'all',
+      showFavourites: false
     };
   },
+  methods: {
+    toggleFavourites() {
+      return (this.showFavourites = !this.showFavourites);
```
Comment (atrunelle) [inline src/App.vue]: This is a bit weird. Why are you returning the result of an assignment? What does it return exactly and where is it used?

File: src/components/Beer.vue [hunk 3]
```diff
@@ -44,6 +55,11 @@ export default {
   props: {
     beer: Object
   },
+  methods: {
+    isBeerFavourite() {
+      return (this.beer.isFavourite = !this.beer.isFavourite);
```
Comment (atrunelle) [inline src/components/Beer.vue]: should this not be `===`? Are you assigning a new value on purpose?

If you are just returning a boolean, without using arguments, a computed property is more performant

File: src/components/Beer.vue [hunk 4]
```diff
@@ -19,6 +19,17 @@
         </div>
 
         <p class="beer__description" data-test="beer-description">{{ beer.description }}</p>
+
+        <button
```
Comment (atrunelle) [inline src/components/Beer.vue]: Same comment as above about toggling text and repeating code

**Review by atrunelle — CHANGES_REQUESTED** [review]
A few changes required as per comments.

Also, it would be nice if the favourites will stay saved when reloading the page. Anything you can think of to make that happen?

File: src/App.vue (hunk 1 above)
Comment (matt-harris) [reply src/App.vue]: Done - moved to computed property 👍

File: src/components/Beer.vue (hunk 4 above)
Comment (matt-harris) [reply src/components/Beer.vue]: Done - moved to computed property 👍

File: src/components/Beer.vue (hunk 3 above)
Comment (matt-harris) [reply src/components/Beer.vue]: This method is toggling the state of the beer object property isFavourite.

This is then used to add a conditional class ```:class="{ 'beer--favourite': beer.isFavourite }"```

And use in the computed property for the button text

```
    beerFavouriteText() {
      return this.beer.isFavourite
        ? 'Remove from favourites'
        : 'Add to favourites';
    }
```

File: src/App.vue (hunk 2 above)
Comment (matt-harris) [reply src/App.vue]: This method is toggling the state of the data object property showFavourites.

This is then used in the filter

```
        .filter(beer => {
          if (this.showFavourites) {
            return beer.isFavourite;
          }

          return beer;
        });
```

And use in the computed property for the button text

```
    showFavouriteText() {
      return this.showFavourites ? 'Show all beers' : 'Show favourites';
    }
```

File: src/App.vue [hunk 5]
```diff
@@ -66,9 +68,20 @@ export default {
...
-      beerStrength: 'all'
+      beerStrength: 'all',
+      showFavourites: false
     };
   },
+  methods: {
+    /**
+     * Toggles the state of showFavourites.
+     *
+     * @returns {Boolean} The state of showFavourites.
```
Comment (atrunelle) [inline src/App.vue]: Just saw this comment can be removed

File: src/components/Beer.vue [hunk 6]
```diff
@@ -44,6 +46,16 @@ export default {
   props: {
     beer: Object
   },
+  methods: {
+    /**
+     * Toggles the state of isFavourite.
+     *
+     * @returns {Boolean} The state of isFavourite.
```
Comment (atrunelle) [inline src/components/Beer.vue]: same for the comment

**Review by atrunelle — APPROVED** [review]
Just need to remove the comments about the return your removed

File: src/App.vue (hunk 5 above)
Comment (matt-harris) [reply src/App.vue]: Return comment removed.

File: src/components/Beer.vue (hunk 6 above)
Comment (matt-harris) [reply src/components/Beer.vue]: Return comment removed.

**Comment by matt-harris** [comment]
Thanks for the great feedback as always @atrunelle

**Pull request merged**